pkg/analyzer/analyzers/airtable/airtablepat/airtable.go

```diff
@@ -0,0 +1,205 @@
+
+	var basesInfo *common.AirtableBases
+	basesReadPermission := common.SchemaBasesRead
+	if granted, err := determineScope(token, basesReadPermission, nil); granted {
+		if err != nil {
+			color.Red("[x] Error : %s", err.Error())
+			return
+		}
+		basesInfo, _ = common.FetchAirtableBases(token)
+		err := determineScopes(token, basesInfo)
+		if err != nil {
+			color.Red("[x] Error : %s", err.Error())
+			return
+		}
+	}
+
+	color.Green("[!] Valid Airtable Personal Access Token\n\n")
+
+	common.PrintUserAndPermissions(userInfo, scopeStatusMap)
+	if scopeStatusMap[common.PermissionStrings[basesReadPermission]] {
+		common.PrintBases(basesInfo)
+	}
+}
+
+func determineScope(token string, scope common.Permission, ids map[string]string) (bool, error) {
+	scopeString := common.PermissionStrings[scope]
+	endpoint := getEndpointByPermission(scopeString)
+	url := endpoint.URL
+	if ids != nil {
+		for _, key := range endpoint.RequiredIDs {
+			if value, ok := ids[key]; ok {
+				url = strings.Replace(url, fmt.Sprintf("{%s}", key), value, -1)
+			}
+		}
+	}
+
+	resp, err := common.CallAirtableAPI(token, endpoint.Method, url)
+	if err != nil {
+		return false, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		scopeStatusMap[scopeString] = true
+		return true, nil
+	} else if endpoint.ExpectedErrorResponse != nil {
+		var result map[string]interface{}
+		if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
+			return false, err
+		}
+
+		if errorInfo, ok := result["error"].(map[string]interface{}); ok {
+			if errorType, ok := errorInfo["type"].(string); ok && errorType == endpoint.ExpectedErrorResponse.Type {
+				scopeStatusMap[scopeString] = false
+				return false, nil
+			}
+		}
+	}
+
+	scopeStatusMap[scopeString] = true
+	return true, nil
+}
+
+func determineScopes(token string, basesInfo *common.AirtableBases) error {
+	if basesInfo != nil && len(basesInfo.Bases) > 0 {
+		for _, base := range basesInfo.Bases {
+			if base.Schema != nil && len(base.Schema.Tables) > 0 {
+				ids := map[string]string{"baseID": base.ID}
+				tableScopesDetermined := false
+
+				// Verify token "webhooks:manage" permission
+				_, err := determineScope(token, common.WebhookManage, ids)
+				if err != nil {
+					return err
+				}
+				// Verify token "block:manage" permission
+				_, err = determineScope(token, common.BlockManage, ids)
+				if err != nil {
+					return err
+				}
+
+				// Verifying scopes that require an existing table
+				for _, table := range base.Schema.Tables {
+					ids["tableID"] = table.ID
+
+					if !tableScopesDetermined {
+						_, err = determineScope(token, common.SchemaBasesWrite, ids)
+						if err != nil {
+							return err
+						}
+						_, err = determineScope(token, common.DataRecordsWrite, ids)
+						if err != nil {
+							return err
+						}
+						tableScopesDetermined = true
+					}
+
+					if granted, err := determineScope(token, common.DataRecordsRead, ids); err != nil {
```

Comment:
suggestion: We can also remove the `if else` and just use an `if` at the same indentation level, since the `if` block already returns.

Ex:
```
granted, err := determineScope(token, common.DataRecordsRead, ids)
if err != nil {
	return err
}
if !granted {
    continue
}
...
```